tests/python/test_dynamic.py

```diff
@@ -135,14 +135,14 @@ def func():
         assert x[i] + 3 == z[i]
 
 
-@test_utils.test(require=ti.extension.sparse)
+@test_utils.test(require=ti.extension.sparse, exclude=[ti.metal])
 def test_dense_dynamic():
     # The spin lock implementation has triggered a bug in CUDA, the end result
     # being that appending to Taichi's dynamic node messes up its length. See
     # https://stackoverflow.com/questions/65995357/cuda-spinlock-implementation-with-independent-thread-scheduling-supported
     # CUDA 11.2 didn't fix this bug, unfortunately.
-    if ti.lang.impl.current_cfg().arch == ti.cuda:
-        pytest.skip('CUDA spinlock bug')
+    # if ti.lang.impl.current_cfg().arch == ti.cuda:
```

Comment:
I tried to run this test on CUDA and it seems alright now. Should we delete those comments and enable this test on CUDA now? @k-ye

Reply:
@lin-hitonami to isolate the changes, can you remove this skip & comment in a separate PR? ;)